docs/spec_draft.md

```diff
@@ -680,6 +681,48 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.if
+
+### Semantics
+
+Returns the result of `true_branch` if `pred` is `true`, `false_branch` if
+`pred` is `false`.
+
+### Inputs
+
+| Name           | Type                                       |
+|----------------|--------------------------------------------|
+| `pred`         | 1-dimensional tensor constant of type `i1` |
+| `true_branch`  | function of type `function`                |
+| `false_branch` | function of type `function`                |
+
+### Outputs
+
+| Name     | Type               |
+|----------|--------------------|
+| `result` | any supported type |
+
+### Constraints
+
+  * (C1) `true_branch` and `false_branch` take 0 arguments.
```

Comment:
At the top we are defining what a function is. IMO, we should say that the function can capture implicit arguments from the parent scope.

Reply:
Later today, I was planning to send a PR that describes function semantics. Will ping it here when done.

Reply:
@sdasgup3 yes, I was thinking about that as well, as in the functions have 0 arguments but they can use whatever is in the scope of parent function.